app/templating/view_context.py

```diff
@@ -88,39 +97,147 @@ def build_view_context_for_question(metadata, schema, answer_store, current_loca
+
+    context = build_view_context_for_summary(schema, section_list, answer_store, metadata,
+                                             csrf_token, block_type, variables)
+
+    context['summary'].update({
+        'title': title,
+    })
+    return context
+
+
+def build_view_context_for_calculated_summary(metadata, schema, answer_store, schema_context, block_type,
+                                              variables, csrf_token, current_location):
+    rendered_block = renderer.render(schema.get_block(current_location.block_id), **schema_context)
+    section_list = _build_calculated_summary_section_list(schema, rendered_block, current_location.group_id)
+
+    context = build_view_context_for_summary(schema, section_list, answer_store, metadata,
+                                             csrf_token, block_type, variables)
+
+    formatted_total = _get_formatted_total(rendered_block, answer_store, current_location.group_instance, schema)
+    context['summary'].update({
+        'calculated_question': _get_calculated_question(rendered_block['calculation'], answer_store, schema,
+                                                        metadata, current_location.group_instance, formatted_total),
+        'title': get_question_title(rendered_block, answer_store, schema, metadata,
+                                    current_location.group_instance) % dict(total=formatted_total),
+    })
+    return context
 
-    elif block_type == 'SectionSummary':
-        for section in schema.json['sections']:
-            group_ids = (group['id'] for group in section['groups'])
-            if current_location.group_id in group_ids:
-                section_list = [renderer.render(section, **schema_context)]
-                title = section['title']
-                break
 
+def build_view_context_for_summary(schema, section_list, answer_store, metadata,
+                                   csrf_token, block_type, variables):
     summary_rendering_context = build_summary_rendering_context(schema, section_list, answer_store, metadata)
 
     context = {
         'csrf_token': csrf_token,
         'summary': {
             'groups': summary_rendering_context,
             'answers_are_editable': True,
-            'is_view_submission_response_enabled': is_view_submitted_response_enabled(schema.json),
             'summary_type': block_type,
-            'title': title,
         },
         'variables': variables,
     }
     return context
 
 
+def _build_calculated_summary_section_list(schema, rendered_block, group_id):
+    blocks = []
+    for block in schema.blocks:
+        answers_in_block = schema.get_answers_by_id_for_block(block['id'])
+        answers_to_calculate = rendered_block['calculation']['answers_to_calculate']
+
+        answer_matches = set(answers_in_block.keys()) & set(answers_to_calculate)
+        if answer_matches:
+            blocks.append(_remove_unwanted_questions_answers(block, answers_in_block, answer_matches))
```

Comment:
is it worth maybe doing something like `blocks = [_remove_unwanted_questions_answers(block, answers_in_block, answer_matches)]`